llama_index/vector_stores/postgres.py

```diff
@@ -331,15 +331,29 @@ def _apply_filters_and_limit(
     ) -> Any:
         import sqlalchemy
 
+        sqlalchemy_conditions = {
+            "or": sqlalchemy.sql.or_,
+            "and": sqlalchemy.sql.and_,
+        }
+
         if metadata_filters:
-            for filter_ in metadata_filters.legacy_filters():
-                bind_parameter = f"value_{filter_.key}"
-                stmt = stmt.where(  # type: ignore
-                    sqlalchemy.text(f"metadata_->>'{filter_.key}' = :{bind_parameter}")
+            if metadata_filters.condition not in sqlalchemy_conditions:
+                raise ValueError(
+                    f"Invalid condition: {metadata_filters.condition}. "
+                    f"Must be one of {list(sqlalchemy_conditions.keys())}"
                 )
-                stmt = stmt.params(  # type: ignore
-                    **{bind_parameter: str(filter_.value)}
+            stmt = stmt.where(
+                sqlalchemy_conditions[metadata_filters.condition](
+                    *(
+                        sqlalchemy.text(
+                            f"metadata_->>'{filter.key}' "
+                            f"{'=' if filter.operator.value == '==' else filter.operator.value} "
```

Comment:
so this is only for exact match? we could support any operator atm.

could we just pass filter.operator and transform to postgres supported operator.
you can follow this way: https://github.com/run-llama/llama_index/blob/5dab77427d26afe25077f6c69370bb046bd30a08/llama_index/vector_stores/chroma.py#L34-L49

Reply:
This is fundamentally what I'm doing; only the "==" needs to be transformed to "="

Reply:
Thanks. The type checking isn't needed because `MetadataFilter.operator` is always a `FilterOperator`.
We also still need the condition (OR/AND).
Here's what I have:
```python
stmt = stmt.where(  # type: ignore
    sqlalchemy_conditions[metadata_filters.condition](
        *(
            sqlalchemy.text(
                f"metadata_->>'{filter_.key}' "
                f"{self._to_postgres_operator(filter_.operator)} "
                f"'{filter_.value}'"
            )
            for filter_ in metadata_filters.filters
        )
    )
)
```
I will test and push in a moment.

Reply:
Hm, guess it is too late.

Reply:
Feel free to make a followup pr if you want to optimize this